src/Runner.Listener/Runner.cs

```diff
@@ -466,6 +466,43 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     Trace.Info("Refresh message received, skip autoupdate since a previous autoupdate is already running.");
                                 }
                             }
+                            else if (string.Equals(message.MessageType, RunnerRefreshMessage.MessageType, StringComparison.OrdinalIgnoreCase))
+                            {
+                                if (autoUpdateInProgress == false)
+                                {
+                                    autoUpdateInProgress = true;
+                                    var runnerUpdateMessage = JsonUtility.FromString<RunnerRefreshMessage>(message.Body);
+                                    var agentRefreshMessage = new AgentRefreshMessage(runnerUpdateMessage.AgentId, runnerUpdateMessage.TargetVersion, TimeSpan.FromMilliseconds(runnerUpdateMessage.Timeout));
```

Comment:
can we merge this into the if the block on 433 so you don't have to duplicate all the code?

basically,
```C#
if (string.Equals(message.MessageType, AgentRefreshMessage.MessageType, StringComparison.OrdinalIgnoreCase) ||
    string.Equals(message.MessageType, RunnerRefreshMessage.MessageType, StringComparison.OrdinalIgnoreCase))
{
   AgentRefreshMessage runnerUpdateMessage = null;
   if (string.Equals(message.MessageType, AgentRefreshMessage.MessageType, StringComparison.OrdinalIgnoreCase))
    {
         runnerUpdateMessage = JsonUtility.FromString<AgentRefreshMessage>(message.Body);
    }
    else
    {
       var brokerRunnerUpdateMessage = JsonUtility.FromString<RunnerRefreshMessage>(message.Body); 
       runnerUpdateMessage = new AgentRefreshMessage(brokerRunnerUpdateMessage.AgentId, brokerRunnerUpdateMessage.TargetVersion, TimeSpan.FromMilliseconds(brokerRunnerUpdateMessage.Timeout));
    }

    // existing code.
}
```